server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+        # Error checking is done on these options in the sync manager before the database is queried
+        entries = sync_manager.sync_history(repo_id, limit=limit, sort=sort, start_date=start_date,
+                                            end_date=end_date)
+        return generate_json_response_with_pulp_encoder(entries)
+
+
+class RepoPublishHistory(View):
+    """
+    View for publish history of a specified repository.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, repo_id, distributor_id):
+        """
+        Retrieve publish history for a specified distributor.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of the repository
+        :type  repo_id: str
+        :param repo_id: retrieve the publish history of this distributor
+        :type  repo_id: str
+
+        :return: Response containing a list of dicts, one for each publish event
+        :rtype : django.http.HttpResponse
+        """
+
+        sort = request.GET.get(constants.REPO_HISTORY_FILTER_SORT)
+        start_date = request.GET.get(constants.REPO_HISTORY_FILTER_START_DATE)
+        end_date = request.GET.get(constants.REPO_HISTORY_FILTER_END_DATE)
+        limit = request.GET.get(constants.REPO_HISTORY_FILTER_LIMIT)
+        if limit:
+            try:
+                limit = int(limit)
+            except ValueError:
+                raise pulp_exceptions.InvalidValue([constants.REPO_HISTORY_FILTER_LIMIT])
+        if not sort:
+            sort = constants.SORT_DESCENDING
+
+        publish_manager = manager_factory.repo_publish_manager()
+        entries = publish_manager.publish_history(repo_id, distributor_id, limit=limit, sort=sort,
+                                                  start_date=start_date, end_date=end_date)
+        return generate_json_response_with_pulp_encoder(entries)
+
+
+class RepoSync(View):
+    """
+    View for syncing a repository.
+    """
+
+    @auth_required(authorization.EXECUTE)
+    @json_body_allow_empty
+    def post(self, request, repo_id):
+        """
+        Dispatch a task to sync a repository.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of the repository to sync
+        :type  repo_id: str
+
+        :raises pulp_exceptions.OperationPostponed: dispatch a sync repo task
+        """
+
+        overrides = request.body_as_json.get('override_config', None)
+
+        # Check for repo existence and let the missing resource bubble up
+        manager_factory.repo_query_manager().get_repository(repo_id)
+        async_result = repo_tasks.sync_with_auto_publish(repo_id, overrides)
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+
+class RepoPublish(View):
+    """
+    View for publishing a repository.
+    """
+
+    @auth_required(authorization.EXECUTE)
+    @json_body_allow_empty
```

Comment:
or, dig into the code and see if there is an option make possible publish for all distributors if you do not specify anything the body.